woocommerce/woocommerce
# Bump WooCommerce Admin version to 2.7.1-rc.1

This PR bumps the bundled WooCommerce Admin version to `2.7.1-rc.1`.  Note that `2.7.0` was skipped since some errant DB migrations were included in `2.6.x` versions that would have prevented `2.7.0` migrations from running.

## Changes since 2.7.0-rc.1

- Fix: Update option list migrations to run on 2.7.1 #7705
- Update: Update marketing task completion logic. #7586
- Add backwards compability for old function format. #7688 (`@woocommerce/data`)

## Testing

### Adjust marketing completion logic #7586

1. Load this branch and start a new store
1. Finish the onboarding flow, and install a marketing plugin through (**Plugins > Add New** NOT the marketing task) You can install MailPoet or Mailchimp.
1. Go to **WooCommerce > Home** and notice how the **Set up marketing tools** shows as uncompleted.
1. Visit the marketing task and notice how your marketing plugin shows as installed (do nothing)
1. Go back to **WooCommerce > Home** and notice how the marking task is complete now.

### Update option list migrations to run on 2.7.1 #7705

1. Install WCA `2.6.4` (or WC `5.7` which includes it)
2. Hide the task list via the dropdown
3. Upgrade to this version/branch
4. Note that the task list is still hidden and stays hidden on refresh
5. Optionally, not the option has been migrated to `woocommerce_task_list_hidden_lists`

### Add old option back with depreciated notice for backwards compatibility #7688

1. Detailed test instructions:
1. Install Composite Products plugin can be found under: https://woocommerce.com/my-account/downloads/
1. Create a composite product and make sure the analytics tables are updated ( run action scheduler )
1. Go to Analytics > Composites
1. The table should load fine, and notice how there is a depreciation notice in the console.
